bindings/java/src/org/sleuthkit/datamodel/WindowsAccountUtils.java

```diff
@@ -171,26 +266,173 @@ static boolean isWindowsSpecialSid(String sid) {
 	}
 	
 	/**
-	 * Get the name for the given special Windows SID.
+	 * Checks if the given realmName/loginName is a well known account..
 	 * 
 	 * @param sid SID to check.
 	 * 
-	 * @return Name for Windows special SID, an empty string if the SID is not a known special SID. 
+	 * @return True if the SID is a Windows well known SID, false otherwise 
+	 */
+	static boolean isWindowsWellKnownAccountName(String loginName, String realmName) {
+		
+		String resolvedRealmName = toWellknownEnglishRealmName(realmName);
+		String resolvedLoginName = toWellknownEnglishLoginName(loginName);
+		if (StringUtils.isBlank(resolvedRealmName) ||  StringUtils.isBlank(resolvedLoginName)) {
+			return false;
+		}
+		
+		return SPECIAL_ACCOUNTS_TO_SID_MAP.contains(resolvedRealmName.toUpperCase(), resolvedLoginName.toUpperCase());
+		
+	}
+	
+	/**
+	 * Get the realm address for the given well known Windows SID.
+	 * 
+	 * @param sid SID to check.
+	 * @return Realm Name for Windows special SID, an empty string if the SID is not a known special SID. 
+	 * 
+	 * @throws TskCoreException 
 	 */
-	static String getWindowsSpecialSidName(String sid) {
+	private static String getWindowsWellKnownSidRealmAddr(String sid) throws TskCoreException {
+		String tempSID = stripWindowsBackupPostfix(sid);
+
+		if (SPECIAL_SIDS_MAP.containsKey(tempSID)) {
+			return SPECIAL_SIDS_MAP.get(tempSID).getRealmAddr();
+		}
+		
+		for (Entry<String, WellKnownSidInfo> specialPrefixEntry : SPECIAL_SID_PREFIXES_MAP.entrySet()) {
+			if (tempSID.startsWith(specialPrefixEntry.getKey())) {
+				return specialPrefixEntry.getValue().getRealmAddr();
+			}
+		}
+
+		Matcher match = WINDOWS_SPECIAL_ACCOUNT_PREFIX_REGEX.matcher(tempSID);
+		if (match.find()) {
```

Comment:
Is the idea here that if we come across one of the special sids say s-1-5-100 that we currently don't have a realm for we will put in a real of NT AUTHORITY-100 as a placeholder? This would get updated later if the account ever came in with the real domain?

Reply:
We are returning the realm address (SID) here.  So we will return "S-1-5-100" in this case.